docs-source/src/architecture.rst

```diff
@@ -42,33 +42,484 @@ Thus, each license has a ``grace time`` period that is used to indicate how long
 After the ``grace time`` expires, the booking is deleted. This means that the license was checked out from the
 license server and doesn't need a booking anymore.
 
-License Manager Backend
------------------------
-The ``license-manager-backend`` provides a RESTful API where licenses, bookings and license configurations are tracked.
-The ``license-manager-agent`` uses this API to store the license usage information and to process the booking requests.
+License Manager API
+-------------------
+The ``License Manager API`` provides a RESTful API where licenses and bookins are tracked.
+The ``License Manager Agent`` uses this API to store the license usage information and to process the booking requests.
 
-The backend is also responsible for verifying if the booking requests can be satisfied by accounting for bookings already
+The API is also responsible for verifying if the booking requests can be satisfied by accounting for bookings already
 made and the license usage in the license server.
 
+The API contains 8 entities that have relationship among them.
+This means that some of the resources need to be created before others can be created as well.
+
+.. image:: images/diagram.png
```

Comment:
Do you see a way to refer to where this diagram can be edited? Maybe adding the link to its source as a comment in this markdown file, or adding the source file (like SVG for others) to the repo as well. In this way, it will be easier to update it if needed.

As a side note, many documentation frameworks support [Mermaid](https://mermaid.js.org). It is an excellent way to have the diagram as a code and track changes to it on version control. On the other hand, it can be difficult to fine-tune its design, so sometimes it is not the ideal tool. For instance, see the diagram for the new data model on Jobbergate: https://gist.github.com/fschuch/51006fa5dc18ab24dee8533bf3fefe48

Reply:
This is great suggestion, I looked into adding Mermaid into Sphinx and used it to do a new version of the diagram. Thank you!